sdk/communication/azure-communication-phonenumbers/src/test/java/com/azure/communication/phonenumbers/SipRoutingIntegrationTestBase.java

```diff
@@ -138,24 +135,40 @@ protected SipRoutingClientBuilder getClientBuilderWithConnectionString(HttpClien
     protected SipRoutingClientBuilder getClientBuilderUsingManagedIdentity(HttpClient httpClient) {
         SipRoutingClientBuilder builder = new SipRoutingClientBuilder();
         builder
-            .endpoint(new CommunicationConnectionString(CONNECTION_STRING).getEndpoint())
-            .httpClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient);
+            .httpClient(getHttpClient(httpClient))
+            .endpoint(new CommunicationConnectionString(CONNECTION_STRING).getEndpoint());
 
         if (getTestMode() == TestMode.PLAYBACK) {
             builder.credential(new FakeCredentials());
         } else {
             builder.credential(new DefaultAzureCredentialBuilder().build());
         }
 
-        if (getTestMode() == TestMode.RECORD) {
-            List<Function<String, String>> redactors = new ArrayList<>();
-            redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
-            builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+        if (shouldRecord()) {
+            addRedactors(builder);
         }
 
         return builder;
     }
 
+    private void addRedactors(SipRoutingClientBuilder builder) {
+        List<Function<String, String>> redactors = new ArrayList<>();
+        redactors.add(data -> redact(data, JSON_PROPERTY_VALUE_REDACTION_PATTERN.matcher(data), "REDACTED"));
+        redactors.add(data -> redact(data, UUID_DOMAIN_REDACTION_PATTERN.matcher(data), ""));
+        builder.addPolicy(interceptorManager.getRecordPolicy(redactors));
+    }
+
+    private boolean shouldRecord() {
+        return getTestMode() == TestMode.RECORD;
+    }
+
+    private HttpClient getHttpClient(HttpClient httpClient) {
+        if (httpClient == null || getTestMode() == TestMode.PLAYBACK) {
```

Comment:
Should a null `HttpClient` in LIVE or RECORD test modes mean playback client or an error?

Reply:
Thanks for pointing out. 
It was based on the implementation of the PhoneNumbersIntegrationTestBase in the same package.
[Comment in PhoneNumbers PR about this](https://github.com/Azure/azure-sdk-for-java/pull/27518#discussion_r822151470)
[Comment in PhoneNumbers PR about this](https://github.com/Azure/azure-sdk-for-java/pull/27518/files#r823097918)

Reply:
Got it, since this is following patterns used elsewhere this is good, just wasn't so sure when I initially saw it